src/server/models/videos.js

```diff
@@ -1 +1,45 @@
-module.exports = {};
+const { collection, getDocs, doc, setDoc, updateDoc, deleteDoc } = require('firebase/firestore');
+
+const dbPromise = require('../../config/firebase');
+
+module.exports = {
+  getAllVideos: async () => {
+    try {
+      const db = await dbPromise;
+      const videoRef = collection(db, 'videos');
+      const querySnapshot = await getDocs(videoRef);
+      const documents = querySnapshot.docs.map((doc) => ({ id: doc.id, ...doc.data() }));
```

Comment:
Getting a few linter errors. I believe the doc parameter should be named something else